packages/@glimmer/application/src/components/component-managers/template-only.ts

```diff
@@ -0,0 +1,173 @@
+import { Owner } from '@glimmer/di';
+import { Tag } from '@glimmer/reference';
+import {
+  ComponentManager as VMComponentManager,
+  RuntimeResolver,
+  ComponentCapabilities,
+  CapturedArguments,
+  Option,
+  Invocation,
+  Environment,
+  WithAotStaticLayout,
+  Destroyable,
+  JitRuntimeResolver,
+  AotRuntimeResolver,
+  CompilableProgram,
+  Template,
+  WithJitStaticLayout,
+  ProgramSymbolTable,
+} from '@glimmer/interfaces';
+import { PathReference, CONSTANT_TAG } from '@glimmer/reference';
+import { DEBUG } from '@glimmer/env';
+
+import { RootReference, TemplateOnlyComponentDebugReference } from '../../references';
+import {
+  ComponentDefinitionState,
+  AotComponentDefinition,
+  JitComponentDefinition,
+} from '../component-definition';
+
+export const CAPABILITIES: ComponentCapabilities = {
+  attributeHook: true,
+  createArgs: true,
+  createCaller: false,
+  createInstance: true,
```

Comment:
This should be DEBUG only (though I think there may be a VM bug when this is set to false... unsure if that's fixed now).

Reply:
Setting this to `false` seems to cause the VM to generate invalid bytecode.

Reply:
Yeah so I guess the bug is still there. Do you mind filing an issue on the VM repo?

Reply:
https://github.com/glimmerjs/glimmer.js/issues/215